src/3d/symbols/qgspointcloud3dsymbol.cpp

```diff
@@ -15,46 +15,206 @@
 
 #include "qgspointcloud3dsymbol.h"
 
+// QgsPointCloud3DSymbol
 
-QgsPointCloud3DSymbol::QgsPointCloud3DSymbol()
+QgsPointCloud3DSymbol::QgsPointCloud3DSymbol( QgsPointCloudLayer *layer, QgsPointCloud3DSymbol::RenderingStyle style )
   : QgsAbstract3DSymbol()
+  , mRenderingStyle( style )
+  , mLayer( layer )
```

Comment:
I can't see anywhere that the layer is being used here, and it would be much cleaner to remove this and have symbols as independant from layers. There's also going to be thread safety issues if you DO access the layer here, as the symbols are being used from background threads and it's not safe to access QgsMapLayer's belonging to the main thread on another thread.

Reply:
It is used in https://github.com/qgis/QGIS/blob/768963e2fad3d14e9c49bf75852bc85b6d76b6be/src/3d/qgspointcloudlayerchunkloader_p.cpp#L146
to get the proper attribute since the constructor of QgsPointCloudAttribute takes the size of the attributes and the name and I only have the name,
I guess I can fix this with the use of QgsPointCloudAttributeComboBox